### PR TITLE
feat: Add APIs for custom attribute definitions

### LINKED ISSUE #2655: feat: Custom attribute schema
The custom attributes are rendered as text right now. This is not ideal. We need to allow rich content rendering for the attributes. To support this, add an option to save a custom attribute schema of conversation/contact. The custom attribute schema would define the type of the object which would help the frontend to render the value.

- Create a model to save the custom attribute schema with the following attributes
    - attribute_key - The key which will be used.
    - attribute_type - Let us start with `number`, `text`, `currency`, `percent`, `link`.
    - attribute_display_name - How will it be displayed in the UI.
    - default_value - Provide a default value to the attribute.
    - type_of_object - Contact / Conversation
- CRUD APIs for custom attributes.
- Migration to add the existing custom attributes to the newly constructed schema objects.
    - Create an in-memory array to save the available custom attribute.

### PR DESCRIPTION
**Pull request by pranavrajs**
- Add schema to store custom attribute definition
- Add APIs for CRUD operations on Custom Attribute Definitions

Fixes #2655

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -262,7 +262,7 @@
     t.datetime "agent_last_seen_at"
     t.jsonb "additional_attributes", default: {}
     t.bigint "contact_inbox_id"
-    t.uuid "uuid", default: -> { "gen_random_uuid()" }, null: false
+    t.uuid "uuid", default: -> { "public.gen_random_uuid()" }, null: false
```
Comment (sojan-official) [inline db/schema.rb:265]: intended ?

**Review by sojan-official — COMMENTED** [review]
Should we limit the access to these APIs to administrators alone?

File: db/schema.rb (hunk 1 above)
Comment (pranavrajs) [reply db/schema.rb]: No. Updated it.

**Comment by pranavrajs** [comment]
@sojan-official Please take a look at this.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**